README.md

```diff
@@ -210,3 +210,44 @@ https://jinja.palletsprojects.com/en/3.0.x/templates/#if
 https://pythonbasics.org/flask-login/
 
 https://www.rfc-editor.org/rfc/rfc6749#section-4.1
+
+## module 12 
+
+Installing Terraform 
+
+For step by step instructions to install Terraform, please see the link below:
+
+https://developer.hashicorp.com/terraform/tutorials/azure-get-started/install-cli?in=terraform%2Fazure-get-started
+
+
+Terraform Variables
+
+For keeping your secrets there are more then one way of doing this, here I've setup a file called terraform.tfvars, however as this is a sensitive file it's been added to .gitignore and as such will not be committed, so when using gitpod you'll have to re-create the file everything you start a new session, so for convenience while developing, you can keep the file in something like notepad++ on your local, so you can easily copy and past it in when you start a new session. 
+
+For more information on how to declare the variables please see the followeing page:
+
+https://developer.hashicorp.com/terraform/language/values/variables
+
+
+Create a Service Principal:
+
+https://learn.microsoft.com/en-gb/cli/azure/create-an-azure-service-principal-azure-cli
+
+
+Helpful Commands:
+
+terraform fmt 
+
+This command applies a subset of the Terraform language style conventions, along with other minor adjustments for readability. As a best practice, terraform fmt should always be run on your configuration files, so formatting standards and language style conventions are applied across your configuration in a uniform manner. This helps code readability and ensures all configuration files are formatted the same, no matter which team member is writing the file.
+
+
+terraform -help
```

Comment:
You can use ticks \` to make commands appear as code, making them more readable in Markdown
`terraform -help`, etc.
It is a bit hard to see the commands to run in this text. This would make it much clearer